src/main/java/org/gradle/profiler/BuildActionInvoker.java

```diff
@@ -0,0 +1,68 @@
+package org.gradle.profiler;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.gradle.profiler.Logging.startOperation;
+
+public class BuildActionInvoker {
```

Comment:
This name is a bit too close to `BuildInvoker`. Maybe this could be called `InstrumentedBuild` or `BuildUnderTest`?

Reply:
I'll see if I can come up with some better names.